### PR TITLE
Change the GitHub action to only check for linting and formatting errors [Fixes #1019]

### LINKED ISSUE #1019: Feature Request: Change the GitHub action to only check for Formatting and Linting Errors
**Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is.

**Describe the solution you'd like**
As discussed in [talalwa-admin](https://github.com/PalisadoesFoundation/talawa-admin/pull/437), the GitHub action file should only check for linting and formatting errors, instead of manually linting and formating the code. The same should be implemented here, to maintain consistency across the repositories.
 
**Describe alternatives you've considered**
NA

**Approach to be followed (optional)**
1. Create scripts in `package.json` like `lint:check`, `lint:fix`, `format:check` and `format:fix`.
2. Use these scripts in the GitHub workflows (while ensuring the jobs are configured properly to need other jobs)

**Additional context**
Helps to maintain consistency in the codebase.

### PR DESCRIPTION
**Pull request by EshaanAgg**
**What kind of change does this PR introduce?**
Feature

**Issue Number:**

Fixes #1019

**Did you add tests for your changes?**
NA

**If relevant, did you update the documentation?**
NA

**Summary**
Linted and formatted all the files, modified GitHub workflow to automatically check for linting and formatting errors, and separated all the workflow jobs and configured their order properly with the `needs` argument.

**Does this PR introduce a breaking change?**
No

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**
Yes

### CODE REVIEW

File: .github/workflows/pull-request.yml [hunk 1]
```diff
@@ -20,41 +20,42 @@ env:
...
-        run: npm run lint
-        
+  Type-Checker:
+    name: Checks the code for type errors
+    runs-on: ubuntu-latest
+    needs: Linter
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install Dependencies
+        run: npm ci
```
Comment (palisadoes) [inline .github/workflows/pull-request.yml:39]: 1. Put the type checker in `Linter`like we did in Talawa-adm in.
1. Remove this job and references to it

File: .github/workflows/pull-request.yml [hunk 2]
```diff
@@ -20,41 +20,42 @@ env:
...
       - name: Run Typescript Type-Checker
         run: npm run typecheck 
 
-
   Code-Formatter:
-    name: Formats the code
+    name: Check for formatting errors
     runs-on: ubuntu-latest
-    needs: Linter
+    needs: [Linter, Type-Checker]
```
Comment (palisadoes) [inline .github/workflows/pull-request.yml:46]: This should now just really on Linter

File: .github/workflows/pull-request.yml [hunk 3]
```diff
@@ -20,41 +20,42 @@ env:
...
+      - name: Install Dependencies
+        run: npm ci
+      - name: Run Typescript Type-Checker
+        run: npm run typecheck 
 
   Test-Application:
     name: Testing Application
     runs-on: ubuntu-latest
-    needs: Code-Formatter
+    needs: [Linter, Type-Checker, Code-Formatter]
```
Comment (palisadoes) [inline .github/workflows/pull-request.yml:58]: No need for the type-checker job here

**Review by palisadoes — CHANGES_REQUESTED** [review]

File: .github/workflows/pull-request.yml (hunk 1 above)
Comment (EshaanAgg) [reply .github/workflows/pull-request.yml:39]: Okay!

File: .github/workflows/pull-request.yml (hunk 2 above)
Comment (EshaanAgg) [reply .github/workflows/pull-request.yml:46]: Got it

File: .github/workflows/pull-request.yml (hunk 3 above)
Comment (EshaanAgg) [reply .github/workflows/pull-request.yml:58]: Got it.

**Review by palisadoes — APPROVED** [review]

**Review by xoldyckk — CHANGES_REQUESTED** [review]
@EshaanAgg move typecheck to a seperate job and making testing job dependent on it too.

**Comment by EshaanAgg** [comment]
I had kept TypeChecker as a separate job, but moved it to Linter upon @palisadoes's recommendation to maintain consistency across all repository's.

**Comment by xoldyckk** [comment]
> I had kept TypeChecker as a separate job, but moved it to Linter upon @palisadoes's recommendation to maintain consistency across all repository's. @xoldyckk

Then just chnage it for both talawa-api and talawa-admin. It's not a big deal right?

**Comment by EshaanAgg** [comment]
Same thoughts. 
Should I separate Typechecking into a separate job for both the repositories? 
@palisadoes

**Comment by xoldyckk** [comment]
> Same thoughts. Should I separate Typechecking into a separate job for both the repositories then?

Yeah you should, unless it breaks something which I don't have a clue about?

**Comment by EshaanAgg** [comment]
I don't think so it should. I'll read through all of the workflow files just to be double sure.

**Comment by palisadoes** [comment]
Though it is a separate function:
1. It is related to code quality like linting
1. It repeated the use of the `npm ci` and checkout commands that I thought would slow things down

For clarity, it can be separated again. Make sure that testing only starts after linting, type checking and formatting on both repositories.

**Comment by EshaanAgg** [comment]
Made the required changes @palisadoes @xoldyckk

**Comment by xoldyckk** [comment]
@palisadoes Type-checking is not related to code quality. It's a language requirement. Like how in compiled languages we can't use variables without declaring their type. It's the same thing.

It's not slowing down things because most jobs would run in parallel.

**Review by palisadoes — APPROVED** [review]

**Pull request merged**